### PR TITLE
feat: upgrade solidity version

### PR DESCRIPTION
**Pull request by gas1cent**
Updated the `pragma` everywhere to use a newer version of Solidity, as recommended [by Slither](https://github.com/crytic/slither/wiki/Detector-Documentation#recommendation-72)

### CODE REVIEW

**Review by 0xOneTony — APPROVED** [review]

File: solidity/contracts/Greeter.sol [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: UNLICENSED
-pragma solidity >=0.8.4 <0.9.0;
+pragma solidity >=0.8.18 <0.9.0;
```
Comment (wei3erHase) [inline solidity/contracts/Greeter.sol:2]: is it using anything that disallows it to be compiled with sol 0.8.4?

Comment (gas1cent) [reply solidity/contracts/Greeter.sol:2]: @wei3erHase the template does not have many features and can be compiled with any version, really. But we tend to upgrade to a more recent version after forking the template, and in addition to that, the change would promote the use of a newer compiler. 0.8.4 was 2+ years ago 👴🏻

**Review by 0xGorilla — DISMISSED** [review]

**Comment by 0xng** [comment]
Should we lock the pragma version instead of having it float (in a range)? `0.8.20` introduces `PUSH0` for example, an opcode that is not compatible with other chains as of right now, which perhaps can be a little bit dangerous for someone using the boilerplate, compiling with that version, and deploying incompatible bytecode to a different chain by accident.

**Comment by drgorillamd** [comment]
> Should we lock the pragma version instead of having it float (in a range)? `0.8.20` introduces `PUSH0` for example, an opcode that is not compatible with other chains as of right now, which perhaps can be a little bit dangerous for someone using the boilerplate, compiling with that version, and deploying incompatible bytecode to a different chain by accident.

As it is a boilerplate (ie people shouldn't be importing it, rather build on top of it), I'd do it too -> 0.8.19 perhaps (with Paris in the toml)?

**Comment by 0xng** [comment]
> > Should we lock the pragma version instead of having it float (in a range)? `0.8.20` introduces `PUSH0` for example, an opcode that is not compatible with other chains as of right now, which perhaps can be a little bit dangerous for someone using the boilerplate, compiling with that version, and deploying incompatible bytecode to a different chain by accident.
> 
> As it is a boilerplate (ie people shouldn't be importing it, rather build on top of it), I'd do it too -> 0.8.19 perhaps (with Paris in the toml)?

Oh, that hadn't come to mind! Yes, imo updating to perhaps `0.8.19` or `0.8.20` but adding `paris` in the `toml` makes a lot of sense as PUSH0 was introduced in shangai.

**Review by wei3erHase — APPROVED** [review]
👍

**Review by 0xng — APPROVED** [review]

**Review by 0xGorilla — APPROVED** [review]

**Pull request merged**